recipe/activate.sh

```diff
@@ -22,18 +22,34 @@ if [ ! -w "$WRAPPER_DIR" ]; then
     return
 fi
 
-# there's only one lib in that folder, but the libname changes
+# Everything that follows amounts to very carefully creating the
+# symlink if need be.
+
+# there's only one lib in the placeholder folder, but the libname changes
 # based on the version so use a loop instead of hardcoding it.
-for f in "$GDB_PREFIX/$PLACEHOLDER/lib/"*.py; do
-    if [ ! -e "$f" ]; then
+for target in "$GDB_PREFIX/$PLACEHOLDER/lib/"*.py; do
+    if [ ! -e "$target" ]; then
         # If the file doesn't exist, skip this iteration of the loop.
         # (This happens when no files are found, in which case the
         # loop runs with f equal to the pattern itself.)
         continue
     fi
-    target="$WRAPPER_DIR/$(basename "$f")"
-    # We have write permissions to WRAPPER_DIR but not necessarily target.
-    # Thus it's safest to delete the target in case it already exists.
-    rm -f "$target"
-    ln -s "$f" "$WRAPPER_DIR/$(basename "$f")"
+    symlink="$WRAPPER_DIR/$(basename "$target")"
+    # Check if symbolic link already exists and points to correct file
+    if [ -L "$symlink" ] && [ "$(readlink "$symlink")" = "$target" ]; then
+        # Stop if it does
+        continue
+    fi
+    # We have write permissions to WRAPPER_DIR but not necessarily the symlink.
+    # Thus it's safest to delete the symlink in case it already exists.
+    rm -f "$symlink"
+    # Check if symlink still exists after trying to delete it
+    if [ -e "$symlink" ]; then
```

Comment:
Instead of `rm + existence check + ln -s` you can create a new unique link with random name and `mv` to the final place to make it atomic. In two parallel processes `rm` + `-e` can happen the same time and the two `ln -s` will still collide.

Reply:
I modified it. Is that what you had in mind?